.github/workflows/e2e-tests.yml

```diff
@@ -228,19 +213,25 @@ jobs:
           password: ${{ secrets.DOCKERHUB_TOKEN }}
           maildev: true
 
-      - name: Download Metabase enterprise uberjar
-        uses: ./.github/actions/e2e-download-uberjar
+      - name: Retrieve uberjar artifact for ${{ matrix.edition }}
+        uses: actions/download-artifact@v3
         with:
-          edition: 'ee'
-          was-built: ${{ needs.build.result == 'success' }}
+          name: metabase-ee-${{ github.event.pull_request.head.sha || github.sha }}-uberjar
```

Comment:
What are the cases when `github.event.pull_request.head.sha` is missing so we need `|| github.sha`?

Reply:
@iethree suggested we start including the commit SHA1 in the artifact name.

GitHub has a semi-hidden quirk that the HEAD (or the github.sha) points to the merge commit it creates on the fly in the event of `pull_request`, while it points to the actual latest commit in your branch in the event of `push`.

The way to get the latest commit in a branch for pull request is `github.event.pull_request.head.sha`. If that doesn't exist (when we're on `master`, it will simply fall back to the regular `github.sha`.